### PR TITLE
[ML] changing to not use global bulk indexing parameters in conjunction with add(object) calls

### PR DESCRIPTION
**Pull request by benwtrent**
Global parameters, outside of the global index, are ignored for internal callers in certain cases.
If the internal caller is adding requests via the following methods:
```
- BulkRequest#add(IndexRequest)
- BulkRequest#add(UpdateRequest)
- BulkRequest#add(DocWriteRequest)
- BulkRequest#add(DocWriteRequest[])
```
It is better to specifically set the desired parameters on the requests before they are added
to the bulk request object.

This commit addresses this issue for the ML plugin

### CODE REVIEW

File: x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/JobResultsProviderIT.java [hunk 1]
```diff
@@ -684,6 +686,9 @@ public void testGetSnapshots() {
     }
 
     public void testGetAutodetectParams() throws Exception {
+        PlainActionFuture<Boolean> future = new PlainActionFuture<>();
+        createStateIndexAndAliasIfNecessary(client(), ClusterState.EMPTY_STATE, new IndexNameExpressionResolver(), future);
+        future.get();
```
Comment (droberts195) [inline x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/JobResultsProviderIT.java:691]: I think you should move this into the `indexQuantiles` method, so it's closer to the reason why it's needed.  Without comments it's very hard to see why this test needs this and the others don't.

File: x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/AutodetectResultProcessorIT.java [hunk 2]
```diff
@@ -166,6 +169,9 @@ protected void updateModelSnapshotOnJob(ModelSnapshot modelSnapshot) {
         };
         waitForMlTemplates();
         putJob();
+        PlainActionFuture<Boolean> future = new PlainActionFuture<>();
```
Comment (droberts195) [inline x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/AutodetectResultProcessorIT.java:172]:
```suggestion
        // In production opening a job ensures the state index exists. These tests
        // do not open jobs, but instead feed JSON directly to the results processor.
        // A a result they must create the index as part of the test setup. Do not
        // copy this setup to tests that run jobs in the way they are run in production.
        PlainActionFuture<Boolean> future = new PlainActionFuture<>();
```

**Review by droberts195 — APPROVED** [review]
LGTM

It's great to finally find the subtle edge cases that were causing the concrete indices to be created.

My comments are about trying to avoid future mistakes rather than correctness of this PR.

**Comment by benwtrent** [comment]
run elasticsearch-ci/bwc
run elasticsearch-ci/default-distro

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
run elasticsearch-ci/packaging-sample-windows

**Pull request merged**